byron/ledger/executable-spec/test/Ledger/UTxO/Properties.hs

```diff
@@ -29,6 +33,39 @@ moneyIsConstant = withTests 200 . property $ do
   (st0, st) <- firstAndLastState <$> forAll (trace @UTXOW 500)
   reserves st0 + balance (utxo st0) === reserves st + balance (utxo st)
 
+-- | Check that there is no double spending
+noDoubleSpending :: Property
+noDoubleSpending = property $ do
+  t <- forAll (trace @UTXOW 500)
+  let
+    UTxOState {utxo = utxo0} = _traceInitState t
+    txs = body <$> traceSignals OldestFirst t
+  when (all (\ti -> dom (txouts ti) ∩ dom utxo0 == empty) txs) $
+    traverse_ (noCommonInputsTxs txs) (zip txs [0 .. ])
+   where
+    noCommonInputsTxs :: MonadTest m => [Tx] -> (Tx, Int) -> m ()
+    noCommonInputsTxs txs (tx, i) =
+      traverse_ (\txj -> txins' txj ∩ txins' tx === empty) (take i txs)
+
+    txins' :: Tx -> Set TxIn
+    txins' = fromList . txins
+
+-- | Check that UTxO is outputs minus inputs
+utxoDiff :: Property
+utxoDiff = property $ do
+  t <- forAll (trace @UTXOW 500)
+  let
+    (utxo0, utxoSt) = (utxo *** utxo) . firstAndLastState $ t
+    txs = body <$> traceSignals OldestFirst t
+  when (all (\ti -> dom (txouts ti) ∩ dom utxo0 == empty) txs) $
+    foldl' union' empty txs ⋪ (utxo0 ∪ allTxOuts txs) === utxoSt
```

Comment:
you could us Set.unions (foldl union empty) as in

  Set.unions ((fromList . txins) <$> txs)

Reply:
That's a fair point. I guess it boils down to what's more readable to someone and/or what is one's preference. I don't have a strong opinion here. Should I change this?